multiqc/modules/ngsderive/ngsderive.py

```diff
@@ -455,3 +481,79 @@ def add_junctions_data(self):
             [the documentation](https://stjudecloud.github.io/ngsderive/subcommands/junction_annotation/).""",
             plot=bargraph.plot([bardata, bardata], cats, pconfig),
         )
+
+    def add_endedness_data(self):
+        # Write data to file
+        self.write_data_file(self.endedness, "ngsderive_endedness")
+
+        general_data = {}
+        rpt_present = False
+        for sample, endedness_data in self.endedness.items():
+            general_data[sample] = {
+                "endedness": endedness_data.get("Endedness"),
+            }
+            if "Reads per template" in endedness_data:
+                rpt_present = True
+
+        general_headers = OrderedDict()
+        general_headers["endedness"] = {
+            "title": "Predicted Endedness",
+            "description": "Predicted library endedness from ngsderive",
+        }
+        self.general_stats_addcols(general_data, general_headers)
+
+        headers = OrderedDict()
+        headers["FyLn"] = {
+            "title": "f+l- count",
+            "description": "Number of reads with the 'first in template' FLAG set and the 'last in template' FLAG unset",
+            "format": "{:,.d}",
+        }
+        headers["FnLy"] = {
+            "title": "f-l+ count",
+            "description": "Number of reads with the 'first in template' FLAG unset and the 'last in template' FLAG set",
+            "format": "{:,.d}",
+        }
+        headers["FnLn"] = {
+            "title": "f-l- count",
+            "description": "Number of reads with the 'first in template' FLAG unset and the 'last in template' FLAG unset",
+            "format": "{:,.d}",
+        }
+        headers["FyLy"] = {
+            "title": "f+l+ count",
+            "description": "Number of reads with the 'first in template' FLAG set and the 'last in template' FLAG set",
+            "format": "{:,.d}",
+        }
+        if rpt_present:
+            headers["RPT"] = {
+                "title": "Reads per template",
+                "description": "Average number of reads per QNAME template",
+                "format": "{:,.4f}",
+            }
+        headers["endedness"] = {
+            "title": "Predicted Endedness",
+            "description": "Predicted library endedness from ngsderive",
+        }
+
+        table_data = {}
+        for sample, endedness_data in self.endedness.items():
+            table_data[sample] = {}
+            table_data[sample]["FyLn"] = endedness_data.get("f+l-")
+            table_data[sample]["FnLy"] = endedness_data.get("f-l+")
+            table_data[sample]["FnLn"] = endedness_data.get("f-l-")
+            table_data[sample]["FyLy"] = endedness_data.get("f+l+")
+            table_data[sample]["RPT"] = endedness_data.get("Reads per template")
```

Comment:
Unfortunately formatting in the tables doesn't work properly when numbers are passed as strings:

```suggestion
            table_data[sample]["FyLn"] = int(endedness_data.get("f+l-"))
            table_data[sample]["FnLy"] = int(endedness_data.get("f-l+"))
            table_data[sample]["FnLn"] = int(endedness_data.get("f-l-"))
            table_data[sample]["FyLy"] = int(endedness_data.get("f+l+"))
            table_data[sample]["RPT"] = float(endedness_data.get("Reads per template"))
```

Would be good to do that for other subcommands as well, including the general stats table